meteor/server/DatabaseCaches.ts

```diff
@@ -124,6 +124,47 @@ export class ReadOnlyCache {
 		}
 		this._deferredAfterSaveFunctions.length = 0 // clear the array
 
+		if (span) span.end()
+	}
+	/**
+	 * Assert that no changes should have been made to the cache, will throw an Error otherwise. This can be used in
+	 * place of `saveAllToDatabase()`, when the code controlling the cache expects no changes to have been made and any
+	 * changes made are an error and will cause issues.
+	 */
+	assertNoChanges() {
+		const span = profiler.startSpan('Cache.assertNoChanges')
+
+		const allDBs: DbCacheWriteCollection<any, any>[] = []
+		_.map(_.keys(this), (key) => {
+			const db = this[key]
+			if (isDbCacheWriteCollection(db)) {
+				allDBs.push(db)
+			}
+		})
+
+		if (this._deferredFunctions.length > 0)
+			throw new Meteor.Error(
```

Comment:
Should the throws be guarded behind an `if (!Meteor.isProduction)` , to not cause issues in prod?

Reply:
Yeah, I didn't notice that the error for "saveAllToDatabase never called" is only logged when not in production. I would still like to print out something to the log if this assertion is failed, even in production, since that will help us track any bugs that happen in production.